suave/builder/api/api_server.go

```diff
@@ -0,0 +1,43 @@
+package api
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+// sessionManager is the backend that manages the session state of the builder API.
+type sessionManager interface {
+	NewSession() (string, error)
+	AddTransaction(sessionId string, tx *types.Transaction) (*types.SimulateTransactionResult, error)
+}
+
+func NewServer(s sessionManager) *Server {
+	api := &Server{
+		sessionMngr: s,
+	}
+	return api
+}
+
+type Server struct {
+	sessionMngr sessionManager
+}
+
+func (s *Server) NewSession(ctx context.Context) (string, error) {
```

Comment:
wonder if we should consider some sort of sane session cap, additionally, should we add metrics to this maybe so we can observe?

Reply:
We could address this in a followup PR.

Reply:
#165